backend/de.metas.business/src/main/java/de/metas/forex/ForexContract.java

```diff
@@ -171,4 +178,18 @@ public BigDecimal getCurrencyRate(final CurrencyId fromCurrencyId, final Currenc
 		}
 	}
 
+	public void validateSectionCode(@Nullable final SectionCodeId documentSectionCodeId)
+	{
+		if (documentSectionCodeId == null || sectionCodeId == null)
+		{
+			return;
+		}
+
+		if (!this.sectionCodeId.equals(documentSectionCodeId))
+		{
+			throw new AdempiereException("ForexContract.SectionCode ({0}) shall match the Document.SectionCode ({1})",
```

Comment:
pls use AD_Message

Reply:
even if it's not a user validation error?